evidence/services.go

```diff
@@ -0,0 +1,11 @@
+package evidence
+
+import (
+	"github.com/tendermint/tendermint/types"
+)
+
+//go:generate mockery --case underscore --name BlockStore
+
+type BlockStore interface {
+	LoadBlockMeta(height int64) *types.BlockMeta
+}
```

Comment:
At some point we should make `store.BlockStore` an interface; we're already defining these wrapper interfaces in several other places as well, e.g. the blockchain reactors. Maybe create an issue for this?

Reply:
Sure

Reply:
I do think the approach of every reactor/module defining there own interface of expected functions is a good way to go. this does mean that there will be interfaces defined multiple times but each reactor/module should be/try to be isolated. This may go into defining the API's for modules.

Reply:
This incur additional engineering overhead - what's the benefit?

Reply:
I think marko's idea is more intuitive in that each module defines the resources that it needs

Reply:
every module defines what it expects. this way there wont be a reliance on a different module but just the interface. What is the engineering overhead?

This is done in the sdk for a modular system. Tendermint does follow a lot of the same philosophy but lost it along the way.

Reply:
> What is the engineering overhead?

If every module defines its own interfaces for all its dependencies then the number of interfaces scales as n^2. The engineering overhead is in keeping these interfaces updated across all the modules. If we have a block store interface, why should we redeclare that interface 20 times?

Reply:
Another way of looking at it is if we wanted to replace the module with something else we would have to completely adhere to the interface that the blockstore provides not the one that the evidence needs

Reply:
Sure, but in that case I'd argue the interface shouldn't be called `BlockStore`, it should be called e.g. `BlockMetaProvider`.

Reply:
> If every module defines its own interfaces for all its dependencies then the number of interfaces scales as n^2. The engineering overhead is in keeping these interfaces updated across all the modules. If we have a block store interface, why should we redeclare that interface 20 times?

I do agree that is a small overhead added, but for defining a stable API for each module/reactor I think its worth it since we should not be changing these interfaces often. The part of I disagree with is the updating of all the interfaces. If the interface of a module changes, then this does not mean that the other modules need updating.

Reply:
created an issue where we could discuss things: https://github.com/tendermint/tendermint/issues/5220